### DISCUSSION TITLE
Experimental USD support: post screenshots, report problems

### DISCUSSION THREAD

**Discussion by tellypresence** [assimp/assimp#5666] [Show and tell]
Now that USD (`.usd/.usda/.usdc/.usdz`) model format is experimentally supported via `tinyusdz` project, would be great if people could build their viewer implementations with USD support enabled and post screenshots/report problems with models encountered "in the wild"

**Comment by tellypresence** [comment]
USD models that work well
<table>
<tr><th>Model</th><th>Screenshots</th><th>Notes</th></tr>
<tr><td><a href="https://sketchfab.com/3d-models/mitsubishi-f-2-fighter-jet-free-d3d7244554974f499b106e6c11fe3aaf#download"><tt>Mitsubishi F-2 - Fighter Jet - Free</tt></a></td><td><img src="https://github.com/user-attachments/assets/c3d73463-3468-4b7f-8c3c-65d04dc4861f" alt="Mitsubishi F-2 - Fighter Jet - Free" width=180 /><img src="https://github.com/user-attachments/assets/c8f7f69b-d799-4938-bbb7-0f2bdd49936b" alt="Mitsubishi F-2 - Fighter Jet - Free" width=180 /></td><td></td></tr>
<tr><td><a href="https://sketchfab.com/3d-models/dassault-rafale-m-fighter-jet-free-0abf2465adb04bc8bc638688fa54ce21#download"><tt>Dassault Rafale (M) - Fighter Jet - Free</tt></a></td><td><img src="https://github.com/user-attachments/assets/6ac9d911-b133-4778-9c8a-28e5264d0015" alt="Dassault_Rafale_M_-_Fighter_Jet_-_Free" width=180 /><img src="https://github.com/user-attachments/assets/38f62266-4b30-4016-a923-e00b5947845f" alt="Dassault_Rafale_M_-_Fighter_Jet_-_Free" width=180 /></td><td></td></tr>
<tr><td><a href="https://sketchfab.com/3d-models/2010-porsche-918-rsr-race-lab-0b857c510dab4c93a0012a714e3253d1#download"><tt>2010 Porsche 918 RSR (Race Lab)</tt></a></td><td><img src="https://github.com/user-attachments/assets/62cff1b0-0e13-4a8b-96f9-a75e3dd28514" alt="2010_Porsche_918_RSR_Race_Lab" width=180 /><img src="https://github.com/user-attachments/assets/18560f16-2b1f-4525-a633-adb0fb12958e" alt="2010_Porsche_918_RSR_Race_Lab" width=180 /></td><td></td></tr>
<tr><td><a href="https://sketchfab.com/3d-models/robo1-6c2e7a6d42b74e238f758ae6f91a24a6"><tt>Robo1</tt></a></td><td><img src="https://github.com/user-attachments/assets/497bd4fc-273f-4e00-9287-f8cf3bc6e835" alt="Robo1" width=180 /></td><td></td></tr>

<table>

**Comment by AMZN-Gene** [comment]
What app are you using to test? I'm using [assimp-view](https://github.com/assimp/assimp_view), but I don't see the tabs for "Anim" like in your screenshots.
 
I haven't setup loading animations yet, just the rig, so I'm also looking for a way to visualize the bones or bone weights as well.
https://github.com/assimp/assimp/pull/5812

**Reply by tellypresence** [reply]
Using ["cgmv" android app](https://play.google.com/store/apps/details?id=com.tellypresence.cgmv)

**Reply by AMZN-Gene** [reply]
> Using ["cgmv" android app](https://play.google.com/store/apps/details?id=com.tellypresence.cgmv)

Ahh, wouldn't the source code for CGMV be required to test my local changes to assimp?

**Reply by tellypresence** [reply]
Oh right -- no, that project isn't open source

**Reply by AMZN-Gene** [reply]
I see now. Assimp View is also a builtin exe.
https://github.com/assimp/assimp/blob/ce0a50ec21bfa4bb2535a4fbd87b67feadc0cea0/CMakeLists.txt#L753-L758
It supports displaying skeletons
![image](https://github.com/user-attachments/assets/88c77c6b-8069-4ada-b63a-25dad8d2a420)

It wasn't appearing for me in the assimp solution until deleting the check for DirectX. I have DirectX though and the app builds fine :)
https://github.com/assimp/assimp/issues/5820
